samples/hello-oboe/src/main/cpp/PlayAudioEngine.cpp

```diff
@@ -80,10 +84,22 @@ void PlayAudioEngine::createPlaybackStream() {
 
         // TODO: Implement Oboe_convertStreamToText
         // PrintAudioStreamInfo(mPlayStream);
-        prepareOscillators();
+        if (mPlayStream->getFormat() == oboe::AudioFormat::I16){
```

Comment:
After the stream is opened check whether it's 16-bit. 

If it is then create a buffer of floats which we'll render our data into. The buffer size is determined by `stream->getBufferCapacityInFrames` since this is the maximum number of frames we could feasibly be requested to render inside `onAudioReady`